content/05-mysql/10-reading-and-querying-data/05-optimizing-slow-queries.mdx

```diff
@@ -0,0 +1,446 @@
+
+While creating indexes can help you increase the read performance of your datasets, it is important to not over-index your tables.  Indexes come at a cost to write speed, as each time new values are recorded in the table, all of the indexes associated with it must also be updated.  Tables work best if their primary keys are well fitted with the querying patterns that'll be used against them as it allows you to benefit from strong index performance without needing a large number of supplemental indexes to accommodate different access patterns.
+
+### Limiting candidate rows
+
+Another way that you can speed up queries is by limiting the number of candidate and returned rows in your statements when possible.  This is a very context-specific recommendation, so it won't be appropriate in many cases, but if you can pare down the number of rows that MySQL has to evaluate, you can increase performance significantly.
+
+You can do this with the [`LIMIT` clause](https://dev.mysql.com/doc/refman/8.0/en/limit-optimization.html).  For example, to return only the five results, you can use a format like this:
+
+```sql
+SELECT * FROM <table> LIMIT 5
+```
+
+In many cases (depending on what other clauses are included in the query), limiting the number of rows returned can help MySQL short circuit the querying process, stopping sooner than it would normally need to.  This helps save time early on instead of executing over the entire dataset early on and then truncating the results later.
+
+This is especially useful when joining multiple tables.  When at all possible, it is best to limit the results of your query *before* the join occurs.  This allows MySQL to evaluate a small amount of data to the rest of the joins instead of the entirety of the first table.
+
+### Selecting only relevant fields
+
+Another way to increase performance in your queries is to pay attention to what columns you are actually returning.  This can help speed up the results in a number of different ways.
+
+The primary way that limiting the columns you select affects performance is through minimizing network payload.  By returning columns that you don't need, larger queries can substantially increase the network traffic generated by a query.  Selecting only the columns you need to satisfy the logical constraints of a query helps prevent this problem by paring down what is transferred over the network.
+
+Another way that being selective with the columns you return can impact performance is related to indexes.  If MySQL is able to associate all of the columns you're returning with an index, it can potentially operate on that smaller, more organized dataset instead of the full table.  Limiting the columns you are querying allow for this to occur when your indexes match your query patterns well.
```

Comment:
```suggestion
Another way that being selective with the columns you return can impact performance is related to indexes.  If MySQL is able to associate all of the columns you're returning with an index, it can potentially operate on that smaller, more organized dataset instead of the full table.  Limiting the columns you are querying allows for this to occur when your indexes match your query patterns well.
```